railties/lib/rails/commands/server/server_command.rb

```diff
@@ -234,7 +234,17 @@ def host
             options[:binding]
           else
             default_host = environment == "development" ? "localhost" : "0.0.0.0"
-            ENV.fetch("HOST", default_host)
+
+            if env_host = ENV["HOST"]
+              ActiveSupport::Deprecation.warn(<<-MSG.squish)
+                Using the `HOST` environment to specify the IP is deprecated and will be removed in Rails 6.1.
+                Please use `BINDING` environment instead.
+              MSG
```

Comment:
For a smoother upgrade path, I think this should only complain if `BINDING` is not set

Reply:
Good call. I fixed, thanks!